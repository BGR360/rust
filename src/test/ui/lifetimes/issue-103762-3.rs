#![feature(anonymous_lifetime_in_impl_trait)]

trait Trait {}

// Pre-1.64 rejected this because of `+ '_`, 1.64 rejects it because of `-> &str`.
// We should keep rejecting it.
fn d1(_: impl Trait + '_)
    -> &str { loop {} } //~ ERROR missing lifetime specifier

// Pre-1.64 rejected these because of `+ '_`, 1.64 allows them with the feature enabled.
// We should keep allowing them.
fn d2(_: &(impl Trait + '_)) -> &str { loop {} }
fn d3(_: &i32, _: impl Trait + '_) -> &str { loop {} }
fn d4(_: impl Trait + '_, _: &i32) -> &str { loop {} }


fn main() {}
