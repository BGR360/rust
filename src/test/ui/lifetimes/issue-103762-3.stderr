error[E0106]: missing lifetime specifier
  --> $DIR/issue-103762-3.rs:10:8
   |
LL |     -> &str { loop {} }
   |        ^ expected named lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from
help: consider using the `'static` lifetime
   |
LL |     -> &'static str { loop {} }
   |         +++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0106`.
